app/views/moderations/actions_panel.html.erb

```diff
@@ -48,8 +48,14 @@
       <span></span>
       <%= crayons_icon_tag(:checkmark, class: "vomit-checkmark", title: t("views.moderations.actions.checkmark")) %>
     </button>
-    <% if current_user.any_admin? && @moderatable.published %>
-      <% @recent_featured_count = Article.where(featured: true).where("published_at > ?", 1.day.ago).size %>
+    <% if policy(@moderatable).revoke_publication? %>
```

Comment:
We're letting the ArticlePolicy decide if these buttons should be shown, and keeping "unnecessary knowledge" (like `@moderatable.published?`) away from the view.